### DISCUSSION TITLE
Problema com deploy no heroku

### DISCUSSION THREAD

**Discussion by YasmimVieira** [frontendbr/forum#1986] [Perguntas e Respostas]
Olá galera, tudo bem? Queria abrir um discussão aqui para ver se alguém pode me ajudar em algo.
Eu fiz um landing page simples para o portfolio web do meu tio, utilizando apenas html, css, javascript e nodejs para um serviço do formulário de contato.
Quando eu rodo localmente a aplicação, o formulário funciona normalmente, mas quando fiz o deploy no heroku ele parou de funcionar, mas o build ocorreu bem e não tem nenhum erro.
Alguém poderia me ajudar nisso? Não estou conseguindo achar o erro, mas tenho a impressão que o formulário não está conseguindo resgatar os valores, mas não entendo porque acontece isso só quando eu fiz o deploy.

O link do projeto: https://edi-port.herokuapp.com/#
O link do repositório (vocês podem rodar localmente e vão ver que ocorre tudo normalmente):  https://github.com/YasmimVieira/edimar-portfolio

**Comment by gabriel-hahn** [comment]
Olá @YasmimVieira! O que aparece no log lá do Heroku ao enviar o form de contato? É possível ver os logs do projeto e entender o erro que está ocorrendo. Como você só retorna a mensagem padrão 'error' ao Front-end, não consegui identificar o problema por ele, mas como você faz o console.log do error antes, acredito que está aparecendo no Heroku. Tenta validar as tuas variáveis de ambiente lá também, ver se não está faltando preencher alguma variável (`EMAIL_USER`, `USER_PASSWORD`, `EMAIL_TO` e etc). 😃

**Comment by YasmimVieira** [comment]
Você diz o build log? Ou seria outra coisa?
O build log está assim:


-----> Building on the Heroku-20 stack
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NODE_VERBOSE=false
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version 12.x...
       Downloading and installing node 12.20.1...
       Using default npm version: 6.14.10
       
-----> Installing dependencies
       Installing node modules (package.json)
       
       > nodemon@2.0.7 postinstall /tmp/build_53a19345_/node_modules/nodemon
       > node bin/postinstall || exit 0
       
       Love nodemon? You can now support the project via the open collective:
        > https://opencollective.com/nodemon/donate
       
       added 169 packages from 86 contributors and audited 170 packages in 5.592s
       
       11 packages are looking for funding
         run `npm fund` for details
       
       found 0 vulnerabilities
       
       
-----> Build
       
-----> Caching build
       - node_modules
       
-----> Pruning devDependencies
       removed 117 packages and audited 52 packages in 1.602s
       found 0 vulnerabilities
       
       
-----> Build succeeded!
-----> Discovering process types
       Procfile declares types -> web
-----> Compressing...
       Done: 22.9M
-----> Launching...
       Released v7
       https://edi-port.herokuapp.com/ deployed to Heroku



eu configurei todas variáveis de ambiente também

**Comment by gabriel-hahn** [comment]
Existe um botão no canto superior direito chamado 'More' e nele tem a opção 'View logs', la você consegue ver os logs atuais do projeto, não somente o de build, mas de runtime também, que é onde vai aparecer os erros da aplicação durante o uso e todos os logs.
Acessa essa tela de logs, deixa ela aberta, tenta enviar o formulário no teu site em outra aba do navegador e vê se aparece algum erro nessa tela de logs.

![image](https://user-images.githubusercontent.com/19672684/105577535-08709300-5d59-11eb-8ea9-ba9fbc51fc3d.jpeg)
